# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/libwebrtcglue/MediaConduitInterface.h
# Commit: 7d270e96dfbf
# Full Hash: 7d270e96dfbfcc5992f9a1704e7d2229d1a645f3
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 4
# Description:
#   Bug 1654112 - Adapt VideoConduit to the new libwebrtc version. r=ng,bwc
#   
#   This hooks up VideoReceiveStream::Config with a decoder factory and
#   VideoSendStream::Config with both a bitrate allocator factory and an encoder
#   factory.
# ==============================================================================

diff -r 2f1e8f5b2fc2 -r 7d270e96dfbf dom/media/webrtc/libwebrtcglue/MediaConduitInterface.h
--- a/dom/media/webrtc/libwebrtcglue/MediaConduitInterface.h	Mon Feb 08 17:23:44 2021 +0100
+++ b/dom/media/webrtc/libwebrtcglue/MediaConduitInterface.h	Mon Feb 08 18:15:07 2021 +0100
@@ -16,8 +16,10 @@
 #include "mozilla/RefCounted.h"
 #include "TaskQueueWrapper.h"
 #include "VideoTypes.h"
+#include "WebrtcVideoCodecFactory.h"
 
 // libwebrtc includes
+#include "api/video/builtin_video_bitrate_allocator_factory.h"
 #include "api/video/video_frame_buffer.h"
 #include "call/call.h"
 
@@ -267,8 +269,10 @@
         MakeUnique<SharedThreadPoolWebRtcTaskQueueFactory>();
     config.task_queue_factory = taskQueueFactory.get();
     config.trials = aTrials;
+    auto videoBitrateAllocatorFactory = WrapUnique(
+        webrtc::CreateBuiltinVideoBitrateAllocatorFactory().release());
     return new WebRtcCallWrapper(
-        aAudioDecoderFactory,
+        aAudioDecoderFactory, std::move(videoBitrateAllocatorFactory),
         WrapUnique(webrtc::Call::Create(config, aModuleThread)),
         std::move(eventLog), std::move(taskQueueFactory), aTimestampMaker);
   }
@@ -342,12 +346,15 @@
 
  private:
   WebRtcCallWrapper(RefPtr<webrtc::AudioDecoderFactory> aAudioDecoderFactory,
+                    UniquePtr<webrtc::VideoBitrateAllocatorFactory>
+                        aVideoBitrateAllocatorFactory,
                     UniquePtr<webrtc::Call> aCall,
                     UniquePtr<webrtc::RtcEventLog> aEventLog,
                     UniquePtr<webrtc::TaskQueueFactory> aTaskQueueFactory,
                     const dom::RTCStatsTimestampMaker& aTimestampMaker)
       : mTimestampMaker(aTimestampMaker),
         mAudioDecoderFactory(std::move(aAudioDecoderFactory)),
+        mVideoBitrateAllocatorFactory(std::move(aVideoBitrateAllocatorFactory)),
         mEventLog(std::move(aEventLog)),
         mTaskQueueFactory(std::move(aTaskQueueFactory)),
         mCall(std::move(aCall)) {}
@@ -364,6 +371,8 @@
 
  public:
   const RefPtr<webrtc::AudioDecoderFactory> mAudioDecoderFactory;
+  const UniquePtr<webrtc::VideoBitrateAllocatorFactory>
+      mVideoBitrateAllocatorFactory;
   const UniquePtr<webrtc::RtcEventLog> mEventLog;
   const UniquePtr<webrtc::TaskQueueFactory> mTaskQueueFactory;
 